login/token_src.go

```diff
@@ -47,6 +47,7 @@ type OIDCTokenSource struct {
 // NewOIDCTokenSource constructs OIDCTokenSource.
 // Note that OIDC configuration can be passed only from cache. This is due the fact that configuration can be stored in cache as well.
 // If the loginServer is nil, login is disabled.
+// Warning: do not use per request timeouts in ctx. Also use OIDCTokenCtx instead.
 func NewOIDCTokenSource(ctx context.Context, logger *log.Logger, cfg Config, cache Cache, callbackSrv *CallbackServer) (src oidc.TokenSource, clearIDToken func() error, err error) {
```

Comment:
Do NOT pack context into TokenSource struct. We need it in constructor for oidc.NewClient, but NO struct context from now on (: